### PR TITLE
Bluetooth: Mesh: Adding Mesh 1.1 Subnet Bridge

### LINKED ISSUE #74961: Subnetwork bridge for BLE Mesh
There are still not implemented Bluetooth Mesh Protocol v 1.1 features.

Subnetwork bridge is one of them. The feature allows retransmitting frames from one subnetwork to another based on bridging table configured in advance. This allows to better split huge BLE Mesh networks on subnetworks and reduce traffic in general. 

This feature should be implemented as other BLE Mesh features in Zephyr implemented before. There are no special requirements.

The feature is quite huge and requires changing multiple places in existing source code as well as implementation some new functionality like subnetwork bridge table. If do it as at once it will cause an enormous, almost not reviewable final patch bomb. We (some mesh collaborators and I) strictly recommend creating collaboration branch in Zephyr's upstream area. We want to develop it iterationally, using Zephyr's CI to control regression, and involve other Mesh maintainers\collaborators for intermediate PRs review if they have interest. As the final step, we wanted just to create PR to main that was already reviewed and approved multiple times. 

Approximate data for feature development completion is September 2024.

### PR DESCRIPTION
**Pull request by Balaklaka**
Adding the Mesh Protocol v1.1 functionality for the Subnet Bridge functionality.
This PR contain:
* Bridge Configuration Client and Server with API descriptions, implementation and documentation.
* Subnet bridge documentation
* There is both unit tests and bsim test added for the Bridge Configuration Client and Server.
* There is added mesh_shell commands for the Bridge Configuration Client.
* There is added PTS support for the Subnet bridge, Bridge Configuration Client and Server to the tester, and the PTS tests are passing.

Signed-off-by: Ingar Kulbrandstad <ingar.kulbrandstad@nordicsemi.no>
Fixes #74961

### CODE REVIEW

**Comment by Balaklaka** [comment]
Unfortunatelly new Qualification Workspace is not able to properly import ICS if file is UNIX line endings.

**Review by PavelVPV — DISMISSED** [review]

**Comment by jhedberg** [comment]
How is it possible the build and compliance checks are failing? Weren't those run on the collab branch?

**Comment by jhedberg** [comment]
> Unfortunatelly new Qualification Workspace is not able to properly import ICS if file is UNIX line endings.

Ah, this thing again. @nashif @carlescufi once other CI issues have been resolved this will again need a merge override due to the CRLF line endings.

**Comment by PavelVPV** [comment]
> > Unfortunatelly new Qualification Workspace is not able to properly import ICS if file is UNIX line endings.
> 
> Ah, this thing again. @nashif @carlescufi once other CI issues have been resolved this will again need a merge override due to the CRLF line endings.

We created an issue long time ago: https://github.com/zephyrproject-rtos/zephyr/issues/70360. This is closed now, but the problem still exists.

**Comment by alxelax** [comment]
Hi @Balaklaka, there are a couple of clang-format warnings related to coding style. Could you fix them?

**Comment by Balaklaka** [comment]
> Hi @Balaklaka, there are a couple of clang-format warnings related to coding style. Could you fix them?

Fixed, only "Unfortunatelly new Qualification Workspace is not able to properly import ICS if file is UNIX line endings." left

File: tests/bluetooth/tester/src/btp_mesh.c [hunk 1]
```diff
@@ -5163,218 +5154,137 @@ static const struct btp_handler handlers[] = {
...
-		.opcode = BTP_MESH_SAR_RECEIVER_GET,
-		.expect_len = sizeof(struct btp_mesh_sar_receiver_get_cmd),
-		.func = sar_receiver_get
-	},
-	{
-		.opcode = BTP_MESH_SAR_RECEIVER_SET,
-		.expect_len = sizeof(struct btp_mesh_sar_receiver_set_cmd),
-		.func = sar_receiver_set
-	},
+	{.opcode = BTP_MESH_SAR_TRANSMITTER_GET,
```
Comment (PavelVPV) [inline tests/bluetooth/tester/src/btp_mesh.c:5162]: Did you run `clang-format` on these commits? This change and some others don't seem necessary. Did CI complained about these lines?

Comment (Balaklaka) [reply tests/bluetooth/tester/src/btp_mesh.c:5162]: Yes, I started running the "./scripts/ci/check_compliance.py -c 1471c2f81430fd0a381c37d263a4d0ff966e0cb4..HEAD" locally at the end. And fixed all the faults that it complained on, or I would have fixed just a few for every push I did.

**Review by alxelax — APPROVED** [review]

**Review by PavelVPV — APPROVED** [review]

**Comment by PavelVPV** [comment]
@carlescufi, I assume this PR won't show up in the mergelist due to failing CI. How it can be merged? It has 2 approvals now and CI is failing because of line endings in pts files.

**Review by nashif — APPROVED** [review]

**Pull request merged**